src/java/org/apache/cassandra/index/sai/StorageAttachedIndexGroup.java

```diff
@@ -113,6 +137,7 @@ public void invalidate()
     {
         // in case of dropping table, sstable contexts should already been removed by SSTableListChangedNotification.
         queryMetrics.release();
+        groupMetrics.release();
         stateMetrics.release();
         baseCfs.getTracker().unsubscribe(this);
     }
```

Comment:
nit: Might be worth throwing a `@SuppressWarnings("SuspiciousMethodCalls")` on `containsIndex()` to avoid the warning. We know it's not a big deal that other index types might be fed to it and will harmlessly return `false`.